### PR TITLE
Security Accountability + QOL Update

### PR DESCRIPTION
**Pull request by Vivalas**
Adds on to the cell logs functionality of brig timers added recently. Brig timers will now search the datacore for the name of the prisoner, and automatically set their status to incarcerated and add a note to their records.

When released, their status will be set to released.

As well, to enforce accountability with sentencing and help out IAA, brig timers now announce sentences and their arresting officers over the radio, and they also announce when the timer is ended manually and by who.

EDIT: Minor thing but update_all_mob_security_hud() is now a global proc for the sanity of all who want to mess with record code after me.
:cl: Vivalas
add: Brig timers will now update the records of those who are imprisoned in their cell, as long as their name is entered correctly.
add: Sentences and arresting officers are announced over the security channel by brig timers, as well as the names of those who end timers manually.
/:cl:

### CODE REVIEW

File: code/game/machinery/doors/brigdoors.dm [hunk 1]
```diff
@@ -29,32 +29,48 @@
...
+						<b>Duration:</b>		[timetoset/10] seconds<br>
+						<b>Charge(s):</b>	[logcharges]<br>
+						<b>Arresting Officer:</b>		[usr.name]<br><hr><br>
+						<small>This log file was generated automatically upon activation of a cell timer.</small>"}
+
+		playsound(C.loc, "sound/goonstation/machines/printer_dotmatrix.ogg", 50, 1)
+		cell_logs += P
+
+	var/datum/data/record/R = find_security_record("name", logname)
+	Radio.autosay("Detainee [logname] has been incarcerated for [timetoset/10] seconds for the charges of, '[logcharges]'. Arresting Officer: [usr.name].[R ? "" : " Detainee record not found, manual record update required."]", name, "Security", list(z))
```
Comment (SamHPurp) [inline code/game/machinery/doors/brigdoors.dm]: could you break this down with `\` so it isn't one super long line?

**Comment by SamHPurp** [comment]
imo this carries on down the slope of too much automation...

**Comment by Vivalas** [comment]
I think records are one of those things that could benefit from automation, since almost nobody updates them especially on hectic rounds. Records are less of an interesting feature for security but more of a useful tool to help efficiency and maintain sane bookkeeping.

**Comment by Tayyyyyyy** [comment]
I think this is reasonable. It's all too common to have people re-arrested because security forgot to reset their status properly, and having a note in the record for which crimes people have already served time for will speed up processing if they land back in the brig.

**Comment by Vivalas** [comment]
Hm, for some reason the first time someone is incarcerated no logs are put in the records, only the second time.

**Comment by Fruerlund** [comment]
Awesome really wanted to do this myself but didnt know how too!

**Comment by FlattestGuitar** [comment]
Not many people bother with records and it's really helpful for IAAs, other officers and the incarcerated when these are done right. 

I'd support also telling the person behind the bars what they're actually being held for, but otherwise, 👌x💯

**Comment by alexkar598** [comment]
What if there is two persons in the cell at the moment of timer activation?

**Comment by Vivalas** [comment]
Too bad really. The name of the person comes from the name put in the cell log, not the people inside. I can't account for when there are multiple people stacked per cell which is a pretty rare and dangerous thing to begin with.

File: code/game/machinery/doors/brigdoors.dm [hunk 2]
```diff
@@ -29,32 +29,51 @@
 	var/timetoset = 0		// Used to set releasetime upon starting the timer
 	var/obj/item/device/radio/Radio
 	var/printed = 0
+	var/datum/data/record/prisoner
```
Comment (Fox-McCloud) [inline code/game/machinery/doors/brigdoors.dm]: null this out in `Destroy`

**Comment by Vivalas** [comment]
I did the thing merge when?

**Comment by Vivalas** [comment]
@Fox-McCloud Fixed

**Comment by Vivalas** [comment]
🎉

**Pull request merged**